beacon-light-client/plonky2/input_fetchers/publisher/balance-verifier-publisher-lido.ts

```diff
@@ -0,0 +1,163 @@
+import { Redis } from '@dendreth/relay/implementations/redis';
+import { extractHostnameAndPort } from '@dendreth/utils/ts-utils/common-utils';
+import JSONbig from 'json-bigint';
+import { publishTransaction } from '@dendreth/relay/implementations/publish_evm_transaction';
+import Web3 from 'web3';
+import http from 'http';
+import { RequestOptions } from 'https';
+import { ethers } from 'ethers';
+import BalanceVerifierLidoAbi from '../abi/balance_verifier_lido_abi.json';
+
+import { CommandLineOptionsBuilder } from '../utils/cmdline';
+(async () => {
+  const commandOptions = new CommandLineOptionsBuilder()
+    .usage(
+      'Usage: -redis-host <Redis host> -redis-port <Redis port> -take <number of validators>',
+    )
+    .withRedisOpts()
+    .option('rpc-url', {
+      describe: 'The RPC URL',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('gnark-server-url', {
+      describe: 'The URL of the gnark server',
+      type: 'string',
+      demandOption: true,
+    })
+    .withProtocolOpts()
+    .option('transaction-speed', {
+      describe: 'The speed you want the transactions to be included in a block',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('balance-verifier', {
+      describe: 'The address of the balance verifier contract',
+      type: 'string',
+      demandOption: true,
+    })
+    .build();
+
+  const rpcUrl: string = commandOptions['rpc-url'];
+
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+  let privateKey = process.env.USER_PRIVATE_KEY;
+
+  if (privateKey === undefined) {
+    throw new Error('Private key not found');
+  }
+
+  let publisher = new ethers.Wallet(privateKey, provider);
+
+  let balanceVerifierAddress = commandOptions['balance-verifier']!;
+
+  console.log(`Publishing updates with the account: ${publisher.address}`);
+  console.log(`Account balance: ${(await publisher.getBalance()).toString()}`);
+
+  console.log(`Contract address ${balanceVerifierAddress}`);
+
+  const balanceVerifierContract = new ethers.Contract(
+    balanceVerifierAddress,
+    BalanceVerifierLidoAbi,
+    publisher,
+  );
+
+  const web3 = new Web3(rpcUrl);
+
+  if (
+    commandOptions['transaction-speed'] &&
+    !['slow', 'avg', 'fast'].includes(commandOptions['transaction-speed'])
+  ) {
+    throw new Error('Invalid transaction speed');
+  }
+
+  const redis = new Redis(
+    commandOptions['redis-host'],
+    commandOptions['redis-port'],
+  );
+
+  console.log('Publishing proofs');
+
+  let protocol = commandOptions['protocol'];
+
+  redis.subscribeForGnarkProofs(protocol, async () => {
+    console.log('Received new proof');
+    let final_layer_proof_string = (await redis.get(
+      `${protocol}:final_layer_proof`,
+    ))!;
+
+    const final_layer_proof_json = JSON.parse(final_layer_proof_string);
+
+    let final_layer_proof_input = JSON.parse(
+      (await redis.get(`${protocol}:final_proof_input`))!,
+    );
```

Comment:
Either stick to using the constants from kv_db_constants.json or don't define them at all